code/modules/mob/living/simple_animal/hostile/mining/goliath.dm

```diff
@@ -195,3 +252,329 @@
+		speak_emote = list("growls")
+		icon_state = "goliath_subadult"
+		icon_living = "goliath_subadult"
+		icon_dead = "goliath_subadult_dead"
+		ranged = TRUE
+		environment_smash = 2
+		add_draconian_effect()
+	if(growth >= 1200 && subadult == 1 && stat != DEAD) // Grow to adult
+		var/mob/living/simple_animal/hostile/asteroid/goliath/beast/G = new /mob/living/simple_animal/hostile/asteroid/goliath/beast(loc)
+		G.dir = dir
+		if(maxHealth >= 150)
+			G.maxHealth = (G.maxHealth += (maxHealth - 150))
+			G.adjustBruteLoss(maxHealth - 150)
+		if(mind)
+			mind.transfer_to(G)
+		if(tamed == 2)
+			G.faction = list("neutral")
+			G.tamed = TRUE
+		if(draconian)
+			G.draconian = draconian
+			G.name = "draconian goliath"
+			G.add_draconian_effect()
+		qdel(src)
+	if((stat == DEAD) && (tamed != 2)) // So that 1) an ancient goliath can't immediately spawn another juvenile goliath if it dies, and 2) can't easily cheese the juvenile goliath taming process
+		if(prob(10))
+			// 10% chance every cycle to decompose
+			visible_message("<span class='notice'>\The dead body of the [src] decomposes!</span>")
+			gib()
+		if(tamed != 2)
+			tame_progress = 0
+
+/mob/living/simple_animal/hostile/asteroid/goliathjuv/Destroy() // When gibbed / deleted, the ancient goliath that spawned it will be able to spawn another.
+	..()
+	creator.juvgoliathsspawned--
+
+/mob/living/simple_animal/hostile/asteroid/goliathjuv/death()
+	..()
+	add_draconian_effect(draconian_overlay)
+
+/mob/living/simple_animal/hostile/asteroid/goliathjuv/revive()
+	..()
+	add_draconian_effect(draconian_overlay)
+
+/mob/living/simple_animal/hostile/asteroid/goliathjuv/proc/reroll_food() // Picking a random preferred food to eat
+	var/list/food = list(CORE,FLORA,ORGANS,DIAMOND)
+	food_wanted = pick(food)
+
+/mob/living/simple_animal/hostile/asteroid/goliathjuv/proc/becomeaware() // Becoming tamed and player controlled
+	visible_message("<span class='notice'>\The [src] looks around...</span>")
+	spawn()
+		var/list/candidates = pollCandidates("Do you want to play as a tamed young goliath?", ROLE_SENTIENT, 1, 150)
+		if(candidates.len)
+			var/mob/C = pick(candidates)
+			if(C)
+				src.key = C.key
+				to_chat(src, "<span class='biggerdanger'>You are a young goliath!</span>")
+				to_chat(src, "<span class='danger'>You have been, by being fed and treated well, introduced to the non-aggressive way of living, but you have also been disowned by the creatures you previously coexisted with.</span>")
+				picking_candidates = FALSE
+				tamed = 2
+				creator.juvgoliathsspawned--
+				creator = null
+		if(!src.ckey)
+			visible_message("<span class='notice'>\The [src] looks undecided...</span>")
+			tame_progress = 2100
+			picking_candidates = FALSE
+	return
+
+/mob/living/simple_animal/hostile/asteroid/goliathjuv/attackby(obj/item/O, mob/user, params) // Feeding to tame
+	if(istype(O, food_wanted))
+		if(!stat && feed_cooldown <= 0)
+			if(tame_progress <= 599) // Only wants meat
```

Comment:
you already know what kind of food you have, since it has to be of type food_wanted, why do you do these tame_progress checks and ifs? Wouldn't a simple switch statement on food_wanted be much simpler?